src/app/components/tags.ts

```diff
@@ -0,0 +1,36 @@
+import { BaseComponent, type ElementFnProps } from '@components/base-component';
+
+export const span = (props: ElementFnProps<HTMLElement>, ...children: BaseComponent[]) =>
+  new BaseComponent({ ...props, tag: 'span' }, ...children);
+
+export const main = (props: ElementFnProps, ...children: BaseComponent[]) =>
+  new BaseComponent({ ...props, tag: 'main' }, ...children);
+
+export const label = (props: ElementFnProps, ...children: BaseComponent[]) =>
+  new BaseComponent({ ...props, tag: 'label' }, ...children);
```

Comment:
Function label  create BaseComponent without specifying a generic type. Consider adding explicit types for clarity.
```suggestion
export const label = (props: ElementFnProps<HTMLLabelElement>, ...children: BaseComponent[]) =>
  new BaseComponent<HTMLLabelElement>({ ...props, tag: 'label' }, ...children);
```